File: 17-day/order-number.py
def sortNumbers(numbers):
    numbers.sort(reverse = True)
    return numbers


quantity = int(input('How many numbers do you want order: '))
numbers = []
for i in range(0, quantity):
    numbers.append(float(input('Enter the number ({}/{}): '.format(i + 1, quantity))))

print('Numbers orders: {}'.format(sortNumbers(numbers)))


# version 2

import random

if __name__ == "__name__":
    input('Enter any key to generate a list of numbers: ')
    random_list = []
    for i in range(0, 10):
        random_number = random.randint(0, 100)
        random_list.append(random_number)
    print('Your random list: {}' .format(random_list))
    print('Your list in asscending order: {}' .format(sorted(random_list)))
    print('Your list in descending order: {}' .format(sorted(random_list, reverse=1)))

File: 17-day/order-number.js
function sortNumbers(numbers) {
  numbers.sort((a, b) => b - a)
  return numbers
}

const quantity = parseInt(prompt('How many numbers do you want order: '))
const numbers = []
for (let i = 0; i < quantity; i++) {
  numbers.push(parseFloat(prompt(`Enter the number (${i + 1}/${quantity})`)))
}

alert(`Numbers orders: ${numbers}`)

// version 2

function reverseQuickSort(array) {
  let sortedArray = quickSort(array, 0, array.length - 1)
  console.log(sortedArray.reverse());
}

function quickSort(array, left, right) {
  if (left < right) {
    index = quickSortPartition(array, left, right)
    quickSort(array, left, index - 1)
    quickSort(array, index + 1, right)
  }
}

function quickSortPartition(array, left, right) {
  let pivot = array[right]
  let i = let

  for (let j = 0; j < array.length; j++) {
    if (array[j] < pivot) {
      swap(array, i, j)
      i++
    }
  }
  swap(array, i, right)

  return i
}

function swap(array, numA, numB) {
  let aux = array[numA]
  array[numA] = array[numB]
  array[numB] = aux
}